applications/sam2/Dockerfile

```diff
@@ -16,40 +16,60 @@
 # limitations under the License.
 
 
-FROM nvcr.io/nvidia/pytorch:24.06-py3 as pytorch
+FROM nvcr.io/nvidia/pytorch:24.06-py3 AS pytorch
+# FROM nvcr.io/nvidia/pytorch:24.05-py3 as pytorch
 
 
 # Install dependencies
 RUN apt-get update && apt-get install -y \
-    git \
+    git \   
     git-lfs \ 
     x11-apps \
     && rm -rf /var/lib/apt/lists/*
 
 # Set up Git LFS
 RUN git lfs install
 
+
 # Clone sam2
 WORKDIR /workspace
 ARG COMPUTE_CAPACITY
-RUN git clone https://github.com/facebookresearch/segment-anything-2.git \
-    && cd segment-anything-2 \
+RUN git clone https://github.com/facebookresearch/sam2.git \
+    && cd sam2 \
     && python3 -m pip install --no-cache-dir -e . \
     && python3 -m pip install --no-cache-dir -e ".[demo]" \
     && cd checkpoints \
     && ./download_ckpts.sh
+# Add sam2 to PYTHONPATH
+ENV PYTHONPATH="/workspace/sam2"
 WORKDIR /workspace
 
+
+
 # Check the architecture and download the CUDA keyring
 RUN if [ $(uname -m) = "aarch64" ]; then ARCH=arm64 \
     wget https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/sbsa/cuda-keyring_1.1-1_all.deb \
     ; elif [ $(uname -m) = "x86_64" ]; then ARCH=x86_64 \
     wget https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64/cuda-keyring_1.1-1_all.deb \
     ; else echo "Unsupported architecture"; fi
+# RUN dpkg -i cuda-keyring_1.1-1_all.deb \
+#     && apt-get update \
+#     && apt-get -y install holoscan 
+
+# # Persistently set PYTHONPATH
+# ENV PYTHONPATH="/opt/nvidia/holoscan/python/lib:$PYTHONPATH"
+
 RUN dpkg -i cuda-keyring_1.1-1_all.deb \
     && apt-get update \
-    && apt-get -y install holoscan \
-    && export PYTHONPATH="/opt/nvidia/holoscan/python/lib"
+    && mkdir -p /var/lib/holoscan \
+    && apt-get -y install --reinstall --download-only holoscan \
+    && dpkg -x /var/cache/apt/archives/holoscan*.deb /var/lib/holoscan \
```

Comment:
I'm curious about this change, could you comment on why `/var/lib/holoscan` is preferred/required?

Reply:
It's not needed and I changed back to the default installation dir in /opt/nvidia. 
I had a wrong suspicion earlier and chose to install holoscan in an application-specific location instead.